zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java

```diff
@@ -98,8 +101,10 @@ public void serverStarted(Server server) throws Exception {
         if (endpoint == null) {
             final ServerPort activePort = server.activePort();
             assert activePort != null;
-            endpoint = Endpoint.of(server.defaultHostname(),
-                                   activePort.localAddress().getPort());
+            final InetAddress inetAddress = SystemInfo.defaultNonLoopbackIpV4Address();
+            final String ipAddressOrHostname = inetAddress != null ? inetAddress.getHostAddress()
+                                                                   : server.defaultHostname();
+            endpoint = Endpoint.of(ipAddressOrHostname, activePort.localAddress().getPort());
```

Comment:
Sometimes users might want to run Armeria servers on Docker or Kubernetes. 
Then they has to expose a container port like `docker run -p 8080:80`.

I'm not strong, but don't we need to provide a way to customize a service registry port and IP(or hostname) without automatic?

Reply:
We have `ZooKeeperUpdatingListenerBuilder.endpoint()` for that. This is set only when the `endpoint` is not set. 😄

Reply:
Ah. I missed `endpoint == null` check. 😅